sycl/include/sycl/accessor.hpp

```diff
@@ -968,9 +968,14 @@ class __SYCL_SPECIAL_CLASS __SYCL_TYPE(accessor) accessor :
   friend class sycl::ext::intel::esimd::detail::AccessorPrivateProxy;
 
 public:
-  using value_type = DataT;
-  using reference = DataT &;
+  using value_type = typename std::conditional<AccessMode == access::mode::read,
+                                               const DataT, DataT>::type;
+  using reference = value_type &;
   using const_reference = const DataT &;
+  using iterator = value_type *;
```

Comment:
Please do not consider my comments as meaningful or blocking since I'm not an expert in this.
But, I'm just curious, will iterator that defined as a pointer work for some weird test cases like 2D sub-acccessor. See Ranged accessors in SYCL 2020 to understand what I mean - https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:accessors.ranged . The phrase that concerns me most is:
> If the ranged accessor is multi-dimensional, the sub-range is allowed to describe a region of memory in the underlying buffer that is not contiguous in the linear address space.

Reply:
Make sense. Need to investigate, thank you

Reply:
Removed this part of changes because it will be done in scope of https://github.com/intel/llvm/pull/6815